### PR TITLE
7.32d update for hero level up and item choices

### PR DESCRIPTION
**Pull request by ArcherWayne**
7.32d changelog
1. bots now can correctly use bloodstone and eternal shroud. 
2. updated the ability/talent levelup and the item choice for the following heros:
   1. razor
   2. viper
   3. sniper
   4. juggernaut
   5. zeus
   6. huskar
   7. phantom assassin
   8. crystal maiden
   9. ember spirit
   10. centaur warrunner
   11. venomancer
   12. doom bringer
   13. slardar
   14. silencer
   15. legion commander
   16. riki
   17. clinkz
   18. leshrac
   19. drow ranger
   20. dragon knight
   21. shadow fiend
   22. axe
   23. bloodseeker
   24. bristleback
   25. spectre
   26. anti-mage
   27. facelss void
   28. bounty hunter
   29. pudge
   30. witch docter
   31. bounty hunter
   32. disrupter
   33. enchantress
   34. magnus
   35. monkey king
   36. weaver
   37. phantom lancer
   38. timbersaw
   39. tusk

### CODE REVIEW

**Comment by ArcherWayne** [comment]
git commit --amend

**Comment by adamqqqplay** [comment]
I think it would be better to merge changelog_7.32d.md into changelog_en.txt

File: util/ItemUsageSystem.lua [hunk 1]
```diff
@@ -533,16 +533,44 @@ function M.UnImplementedItemUsage()
...
+--             (npcBot:GetHealth() / npcBot:GetMaxHealth()) < 0.5
+                
+--          then
+--             npcBot:Action_UseAbilityOnLocation(bst, npcBot:GetLocation())
+--             -- utility.DebugTalk('Using blooding Stone')
+--             return
+--         end
+--     end
+
+    -- zjc:test start
```
Comment (adamqqqplay) [inline util/ItemUsageSystem.lua:551]: Please delete unnecessary code in this file

File: hero_selection.lua [hunk 2]
```diff
@@ -169,147 +169,129 @@ local hero_pool_default_bot = {
 }
 --recoding implemented bots, using in test.
 local hero_pool_test = {
```
Comment (adamqqqplay) [inline hero_selection.lua:171]: hero_pool_test should not be used in non-test environments.

File: item_purchase_silencer.lua [hunk 3]
```diff
@@ -7,17 +7,18 @@ local ItemPurchaseSystem = dofile(GetScriptDirectory() .. "/util/ItemPurchaseSys
 local ItemsToBuy =
 {
 	"item_tango",
-	"item_tango",
+	-- "item_tango",
 	"item_null_talisman",
-	"item_magic_stick",
+	"item_null_talisman", 
+    "item_magic_stick",
```
Comment (adamqqqplay) [inline item_purchase_silencer.lua:13]: You could format these codes first

File: item_purchase_tusk.lua [hunk 4]
```diff
@@ -11,16 +11,19 @@ local ItemsToBuy =
 	"item_tango",
 	"item_bracer",
 	"item_phase_boots",
-	"item_magic_stick",
+    "item_magic_stick",
```
Comment (adamqqqplay) [inline item_purchase_tusk.lua:14]: You could format these codes first.

**Review by adamqqqplay — CHANGES_REQUESTED** [review]
Thanks for your contribution, but you need to delete unnecessary code and format all code before merging.

**Review by adamqqqplay — APPROVED** [review]

**Comment by adamqqqplay** [comment]
LGTM

**Pull request merged**